src/format/opamSwitch.ml

```diff
@@ -19,12 +19,29 @@ let is_external s =
 
 let external_dirname = "_opam"
 
+let check s =
+  if String.compare s "" = 0 &&
+    let re =
+      Re.(compile @@
+          seq [
+            bol;
+            opt @@ seq [ wordc ; char ':'; set "/\\" ];
+            rep @@ diff any @@ set "<>!`$():";
```

Comment:
```suggestion
            rep1 @@ diff any @@ set "<>!`$():";
```
Just making sure we refuse the empty string. This will also refuse Windows disk roots though (`E:\`)

Reply:
First I had `rep1`, but switched to `rep` to be able to parse root switches (`/` or `C:\`), and added the empty string check at the beginning to be sure to refuse empty switches.
Another solution would be to have the regexp : 
``
alt [ seq [bol; seq [ wordc ; char ':'; set "/\\" ]; eol]; (* accept root *)
       seq [bol;                                                                        (* accept non root paths *)
            opt @@ seq [ wordc ; char ':'; set "/\\" ];
            rep1 @@ diff any @@ set "<>!`$():";
             ...
```

Reply:
Ah ok, missed it, all good then :)